src/WebJobs.Script.WebHost/TokenExpiration/TokenExpirationService.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Properties;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.TokenExpiration
+{
+    internal class TokenExpirationService : IHostedService, IDisposable
+    {
+        private readonly IEnvironment _environment;
+        private readonly ILogger<TokenExpirationService> _logger;
+        private readonly IOptionsMonitor<StandbyOptions> _standbyOptionsMonitor;
+        private CancellationTokenSource _cancellationTokenSource;
+        private Task _analysisTask;
+        private bool _disposed;
+        private bool _analysisScheduled;
+
+        public TokenExpirationService(IEnvironment environment, ILogger<TokenExpirationService> logger, IOptionsMonitor<StandbyOptions> standbyOptionsMonitor)
+        {
+            _environment = environment;
+            _logger = logger;
+            _standbyOptionsMonitor = standbyOptionsMonitor;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (_environment.IsCoreTools())
+                {
+                    return Task.CompletedTask;
+                }
+                if (_standbyOptionsMonitor.CurrentValue.InStandbyMode)
+                {
+                    _standbyOptionsMonitor.OnChange(standbyOptions =>
+                    {
+                        if (!standbyOptions.InStandbyMode && !_analysisScheduled)
+                        {
+                            ScheduleTokenExpirationCheck();
+                        }
+                    });
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error starting token expiration service. Handling error and continuing.");
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                _cancellationTokenSource?.Cancel();
+
+                if (_analysisTask != null && !_analysisTask.IsCompleted)
+                {
+                    _logger.LogDebug("Token expiration service stopped before analysis completion. Waiting for cancellation.");
+
+                    return _analysisTask;
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error stopping Token expiration service. Handling error and continuing.");
```

Comment:
I recommend removing this catch statement. Let the caller decide what to do with a failure.